components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -63,8 +70,24 @@ func ignoreNotFound(err error) error {
 // NotebookReconciler reconciles a Notebook object
 type NotebookReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	kubeClient    clientset.Interface
+	Log           logr.Logger
+	Scheme        *runtime.Scheme
+	eventRecorder record.EventRecorder
+}
+
+func NewNotebookReconciler(client client.Client, kubeClient clientset.Interface, log logr.Logger, scheme *runtime.Scheme) *NotebookReconciler {
+	eventBroadcaster := record.NewBroadcaster()
```

Comment:
Since you can get recorder from manager https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L86,
I think you don't need kubeClient.